doc/source/user_guide/style.ipynb

```diff
@@ -102,16 +117,110 @@
+    "Now we can manually link these to our DataFrame using the `Styler.set_table_attributes` and `Styler.set_td_classes` methods (note that table level 'table-cls' is overwritten here by Jupyters own CSS, but in HTML the default text color will be grey)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s = df.style.set_table_attributes('class=\"table-cls\"')\n",
+    "cls = pd.DataFrame(data=[['cls1', None], ['cls3', 'cls2 cls3']], index=[0,2], columns=['A', 'C'])\n",
+    "s.set_td_classes(cls)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The **advantage** of linking to external CSS is that it can be applied very easily. One can build a DataFrame of (multiple) CSS classes to add to each cell dynamically using traditional `DataFrame.apply` and `DataFrame.applymap` methods, or otherwise, and then add those to the Styler. It will integrate with your website's existing CSS styling.\n",
+    "\n",
+    "The **disadvantage** of this approach is that it is not easy to transmit files standalone. For example the external CSS must be included or the styling will simply be lost. It is also, as this example shows, not well suited (at a table level) for Jupyter Notebooks. Also this method cannot be used for exporting to Excel, for example, since the external CSS cannot be referenced either by the exporters or by Excel itself."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Using Table Styles\n",
+    "\n",
+    "Table styles allow you to control broader areas of the DataFrame styling with minimal HTML transfer. Much of the functionality of `Styler` uses individual HTML id tags to manipulate the output, which may be inefficient for very large tables. Using `table_styles` and otherwise avoiding using id tags can greatly reduce the rendered HTML.\n",
+    "\n",
+    "Table styles are also used to control features which can apply to the whole table at once such as greating a generic hover functionality:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s = Styler(df, cell_ids=False, uuid_len=1)\n",
+    "s.set_table_styles([{'selector': 'tr:hover',\n",
+    "                     'props': [('background-color', 'yellow')]}])\n",
+    "s.set_table_styles({\n",
+    "    'A': [{'selector': '',\n",
+    "           'props': [('color', 'red')]}],\n",
+    "    'B': [{'selector': 'td',\n",
+    "           'props': [('color', 'blue')]}]\n",
+    "}, axis=0, overwrite=False)\n",
+    "s.set_table_styles({\n",
+    "    3: [{'selector': 'td',\n",
+    "           'props': [('color', 'green')]}]\n",
+    "}, axis=1, overwrite=False)"
```

Comment:
can you render df with a df.style before and then after you have provided this style update